Task 2: 
- Identify the restaurants with the highest and lowest number of votes.
- Analyze if there is a correlation between the number of votes and the rating of a restaurant.

In [1]:
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns

# Load the data
df = pd.read_csv('/Users/tilakpatel/Personel/Cognifyz intern/code/Data/Dataset.csv', encoding='utf-8')
df = df.dropna(subset=['Cuisines'])

In [14]:
df_votes = df.copy()

In [17]:
df_votes = df_votes.groupby('Restaurant Name')[['Votes', 'Aggregate rating']].agg({'Votes': 'sum', 'Aggregate rating': 'mean'}).sort_values(by='Votes', ascending=False)

In [22]:
top20 = df_votes.head(20)
top20

Votes  Aggregate rating
Restaurant Name                                   
Barbeque Nation            28142          4.353846
AB's - Absolute Barbecues  13400          4.825000
Toit                       10934          4.800000
Big Chill                  10853          4.475000
Farzi Cafe                 10098          4.366667
Truffles                    9682          3.950000
Chili's                     8156          4.580000
Hauz Khas Social            7931          4.300000
Joey's Pizza                7807          4.250000
Peter Cat                   7574          4.300000
Big Yellow Door             7511          4.266667
Saravana Bhavan             7238          4.133333
Starbucks                   7139          3.805556
Pirates of Grill            7091          4.025000
Karim's                     6878          3.030769
Domino's Pizza              6643          2.740506
Subway                      6124          2.907937
Dunkin' Donuts              5974          3.136364
Big Brewsky                 5705          4.500000
Pind Balluchi               5582          2.630000

These are bottom 20 restaurants but whose aggregate ratings and Votes are greater than 0

In [27]:
bottom20 = df_votes[(df_votes['Aggregate rating'] > 0) & (df_votes['Votes'] > 0)].tail(20)
bottom20

Votes  Aggregate rating
Restaurant Name                                           
Droolfi                                4               3.0
Sama Chicken Biryani                   4               2.9
Budapest Kitchen & Bar                 4               3.0
Purani Delhi 6                         4               3.0
Puskar Raj Momos & Chinese Corner      4               3.0
Gayatri's Break Point Restaurant       4               3.1
Aroma                                  4               3.0
GenY Cuisines                          4               3.0
Maa Bhagwati Tiffins                   4               3.0
Rocketchefs                            4               3.1
Khan Kabab Corner                      4               3.0
Dialogue Lounge & Caf��                4               3.0
The Fresh Chicken Store                4               2.9
XLNC Bakers                            4               2.9
Cake2you                               4               3.0
Wow! India                             4               2.7
Chaap Nation                           4               3.0
Lajawab Chinese Food                   4               3.0
Keshar                                 4               3.0
Three Olives                           4               2.9

- Top 20:
    - High Votes (5,500–28,000): Stabilize ratings, making them reliable and reflective of diverse feedback.
    - Ratings Trend: Mostly above 4.0, showing customer satisfaction. Outliers (e.g., Domino's) face unmet expectations despite popularity.
    - Visibility: High exposure drives votes and strengthens market position.

- Bottom 20:
    - Low Votes (4 for all): Ratings (2.7–3.1) are less representative due to limited feedback.
    - Ratings Trend: Tight clustering reflects mediocrity or lack of strong customer impressions.
    - Visibility: Minimal market reach and engagement hinder growth.

- Insights:
    - Votes Stabilize Ratings: Higher votes = reliable ratings; lower votes = variability.
    - Growth Opportunity: Bottom 20 must improve visibility and quality to attract more votes and stabilize ratings.